### ISSUE TITLE
How to train the model?

### ISSUE THREAD

**Issue by qianc62** [chungdz/DRPN#1]
Since the negative feedbacks are considered into the model, tow to train the model? More specifically, what and how to sample negative examples for l_k+1 classification?

**Comment by chungdz** [comment]
The MIND dataset has impressions for each user. Impression is a feedback sequence that has both clicked news and ignored news.   Sample them to be positive history and negative history of one user for model input.

**Comment by qianc62** [comment]
yes, the model's positive & negative feedbacks are sampled from MIND. This is for the input of one user.

My question is that, given the sampled positive & negative feedbacks, the model will produce one logit (Figure 2); how to sample K negative examples for training (i.e., to produce K+1 logits for K+1-way classification)?

**Comment by chungdz** [comment]
For each clicked news to be the positive one, randomly sample k ignored news in the same impression to be the negative ones.

**Comment by qianc62** [comment]
I'm  still not clear about this. According to my understanding, one training example contains N positive feedbacks and N*K negative feedbacks, and produces one-dimension logit (i.e., only a float number: y^ in Fig. 2). But K+1-way classification needs a K+1-dimension logit. Please detail this, thanks!

**Comment by chungdz** [comment]
N positive feedbacks and N negative feedbacks. You can treat it as a multiclassification problem. The target is to find positive news within other k negative news. The similarity score is calculated for 5 news (one positive and four negative). And the model is trained to find the positive one.

**Comment by qianc62** [comment]
Oh I see, I misunderstood the negative feedbacks and negative candidates. Interesting work! Thanks for your answer!
